File: program_data/github_java_sort_function/6/59.java
package sortingAlgorithms.algorithms.linear;

import sortingAlgorithms.algorithms.SortingAlgorithm;

import java.util.ArrayList;
import java.util.List;


public class RadixSort implements SortingAlgorithm {

    
    @Override
    @SuppressWarnings("unchecked")
    public List<Integer> sort(List<Integer> list) {

        
        if (list == null) {
            throw new IllegalArgumentException("ArrayList not specified!");
        }

        
        
        int exp = 1;

        
        int maxDigits = 2;

        for (int i = 0; i < maxDigits; i++) {

            
            List[] bucketList = new ArrayList[10];
            for (int k = 0; k < 10; k++) {
                bucketList[k] = new ArrayList<Integer>();
            }
            for (Integer aList : list) {
                int number = (aList / exp) % 10;
                bucketList[number].add(aList); 
            }
            exp *= 10;
            int index = 0;
            for (int k = 0; k < 10; k++) {
                for (Object num : bucketList[k]) {
                    list.set(index, (Integer) num);
                    index++;
                }
            }
        }
        return list;
    }
}

File: program_data/PKU_raw/86/59.c

int main()
{
	int n, i, j, a[100][60];
	scanf("%d", &n);
	for(i = 0; i < n; i++)
	{
		scanf("%d", &a[i][0]);
		for(j = 1; j <= a[i][0]; j++)
		{
			scanf("%d", &a[i][j]);
		}
		int num = 60;
		if(a[i][0] == 0)
			printf("60\n");
	    for(j = a[i][0]; j >= 1; j--)
		{
			if(a[i][j] + j * 3 <= 60)
			{
				printf("%d\n", 60 - j * 3 );
				break;
			}
			else if(a[i][j] + j * 3 <= 63)
			{
				printf("%d\n", 60 - (j - 1) * 3 - 63 + a[i][j] + j * 3);
				break;
			}
		}
	}
	return 0;
}

